python/hpsmc/batch.py

```diff
@@ -273,45 +327,119 @@ def submit_cmd(self, name, job_params):
         return batch_id
 
 
-class Slurm(Batch):
+class Slurm(BatchSystem):
     """! Submit Slurm batch jobs."""
 
     def __init__(self):
-        os.environ["LSB_JOB_REPORT_MAIL"] = "N"
+
+        super().__init__()
+
+        self.parser.add_argument("-S", "--sh-dir", nargs='?', help="Directory to hold generated shell scripts for Slurm", default=str(Path(os.getcwd(), 'sh')))
 
     def parse_args(self, args):
-        Batch.parse_args(self, args)
-        if self.email:
-            os.environ["LSB_JOB_REPORT_MAIL"] = "Y"
 
-    def build_cmd(self, name, job_params):
-        log_file = os.path.abspath(os.path.join(self.log_dir, 'job.%s.log' % str(name)))
+        cl = super().parse_args(args)
+
+        # Set Slurm scripts dir
+        self.sh_dir = os.path.abspath(cl.sh_dir)
+        logger.info('Slurm sh dir: {}'.format(self.sh_dir))
+        if not os.path.exists(self.sh_dir):
+            os.makedirs(self.sh_dir)
+            logger.info('Created Slurm sh dir: {}'.format(self.sh_dir))
+
+    def default_rundir(self, job_id=None):
+        """!
+        Override the basic implementation for getting the default run directory.
+        """
+        if self.site == 'slac':
+            run_dir = '$LSCRATCH'
```

Comment:
We do not generically want to run in LSCRATCH on Slurm at slac/sdf. This should be configurable.

Reply:
Reading more this might be configurable and will only default to $LSCRATCH? Does this sound right?

Reply:
Yes, it's configurable and this is just a reasonable default.